File: app/src/main/java/com/iavariav/kbmonline/ui/atasan/adapter/MobilAdapter.java
package com.iavariav.kbmonline.ui.atasan.adapter;

import android.content.Context;
import android.content.SharedPreferences;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.iavariav.kbmonline.R;
import com.iavariav.kbmonline.helper.Config;
import com.iavariav.kbmonline.model.MobilModel;

import java.util.ArrayList;

public class MobilAdapter extends RecyclerView.Adapter<MobilAdapter.ViewHolder> {
    private ArrayList<MobilModel> mobilModels;
    private Context context;

    private String id;

    public MobilAdapter(ArrayList<MobilModel> mobilModels, Context context) {
        this.mobilModels = mobilModels;
        this.context = context;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_mobil, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder holder, final int position) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(Config.SHARED_PREF_NAME, Context.MODE_PRIVATE);
        id = sharedPreferences.getString(Config.SHARED_PREF_ID, "");
        holder.tvPlatMobilStatus.setText(mobilModels.get(position).getPLATMOBIL() + " | " + mobilModels.get(position).getSTATUSMOBIL());
        holder.tvJenisMobil.setText(mobilModels.get(position).getJENISMOBIL());
        holder.tvSopirMobil.setText(mobilModels.get(position).getNAMASUPIR());
        holder.tvTypeMobil.setText(mobilModels.get(position).getTYPEMOBIL());
        holder.tvDeskripsiMobil.setText(mobilModels.get(position).getDESKRIPSIMOBIL());

    }


    @Override
    public int getItemCount() {
        return mobilModels.size();
    }
    public class ViewHolder extends RecyclerView.ViewHolder {

        private TextView tvPlatMobilStatus;
        private TextView tvJenisMobil;
        private TextView tvTypeMobil;
        private TextView tvSopirMobil;
        private TextView tvDeskripsiMobil;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvPlatMobilStatus = itemView.findViewById(R.id.tv_plat_mobil_status);
            tvJenisMobil = itemView.findViewById(R.id.tv_jenis_mobil);
            tvTypeMobil = itemView.findViewById(R.id.tv_type_mobil);
            tvSopirMobil = itemView.findViewById(R.id.tv_sopir_mobil);
            tvDeskripsiMobil = itemView.findViewById(R.id.tv_deskripsi_mobil);
        }
    }
}

File: app/src/main/java/com/iavariav/kbmonline/ui/atasan/adapter/MobilAdapter.kt
package com.iavariav.kbmonline.ui.atasan.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.FragmentActivity
import androidx.recyclerview.widget.RecyclerView

import com.iavariav.kbmonline.R
import com.iavariav.kbmonline.helper.Config
import com.iavariav.kbmonline.model.MobilModel

import java.util.ArrayList

class MobilAdapter(private val mobilModels: ArrayList<MobilModel>, private val context: FragmentActivity?) : RecyclerView.Adapter<MobilAdapter.ViewHolder>() {

    private var id: String? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_mobil, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val sharedPreferences = context?.getSharedPreferences(Config.SHARED_PREF_NAME, Context.MODE_PRIVATE)
        id = sharedPreferences?.getString(Config.SHARED_PREF_ID, "")
        holder.tvPlatMobilStatus.setText(mobilModels[position].platmobil + " | " + mobilModels[position].statusmobil)
        holder.tvJenisMobil.setText(mobilModels[position].jenismobil)
        holder.tvSopirMobil.setText(mobilModels[position].namasupir)
        holder.tvTypeMobil.setText(mobilModels[position].typemobil)
        holder.tvDeskripsiMobil.setText(mobilModels[position].deskripsimobil)

    }


    override fun getItemCount(): Int {
        return mobilModels.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val tvPlatMobilStatus: TextView
        val tvJenisMobil: TextView
        val tvTypeMobil: TextView
        val tvSopirMobil: TextView
        val tvDeskripsiMobil: TextView

        init {
            tvPlatMobilStatus = itemView.findViewById(R.id.tv_plat_mobil_status)
            tvJenisMobil = itemView.findViewById(R.id.tv_jenis_mobil)
            tvTypeMobil = itemView.findViewById(R.id.tv_type_mobil)
            tvSopirMobil = itemView.findViewById(R.id.tv_sopir_mobil)
            tvDeskripsiMobil = itemView.findViewById(R.id.tv_deskripsi_mobil)
        }
    }
}
